src/pages/instances/forms/NetworkForm.tsx

```diff
@@ -111,61 +112,66 @@ or remove the originating item"
+                <b>{isReadOnly ? device.name : "Network"}</b>
               </Label>
             ),
-            value: isReadOnly ? (
-              (formik.values.devices[index] as LxdNicDevice).network
-            ) : (
-              <div className="network-device" key={index}>
-                <div>
-                  <Select
-                    label="Network device"
-                    name={`devices.${index}.network`}
-                    id={`networkDevice${index}`}
-                    onBlur={formik.handleBlur}
-                    onChange={formik.handleChange}
-                    value={
-                      (formik.values.devices[index] as LxdNicDevice).network
-                    }
-                    options={getNetworkOptions()}
-                  />
-                  <Input
-                    label="Network name"
-                    name={`devices.${index}.name`}
-                    id={`networkName${index}`}
-                    onBlur={formik.handleBlur}
-                    onChange={formik.handleChange}
-                    value={(formik.values.devices[index] as LxdNicDevice).name}
-                    type="text"
-                    placeholder="Enter name"
-                  />
-                </div>
-                <div>
-                  <Button
-                    className="delete-device"
-                    onClick={() => removeNetwork(index)}
-                    type="button"
-                    appearance="base"
-                    aria-label="delete network"
-                    hasIcon
-                  >
-                    <Icon name="delete" />
-                  </Button>
+            value:
+              device.type === "custom-nic" ? (
+                "custom network"
+              ) : isReadOnly ? (
+                (formik.values.devices[index] as LxdNicDevice).network
+              ) : (
+                <div className="network-device" key={index}>
+                  <div>
+                    <Select
+                      label="Network device"
+                      name={`devices.${index}.network`}
+                      id={`networkDevice${index}`}
+                      onBlur={formik.handleBlur}
+                      onChange={formik.handleChange}
+                      value={
+                        (formik.values.devices[index] as LxdNicDevice).network
+                      }
+                      options={getNetworkOptions()}
+                    />
+                    <Input
+                      label="Network name"
+                      name={`devices.${index}.name`}
+                      id={`networkName${index}`}
+                      onBlur={formik.handleBlur}
+                      onChange={formik.handleChange}
+                      value={
+                        (formik.values.devices[index] as LxdNicDevice).name
+                      }
+                      type="text"
+                      placeholder="Enter name"
+                    />
```

Comment:
I would still show this input field - as disabled - for custom networks. It's odd that currently we show the network name of a custom network in read mode but we hide it in edit mode.

Reply:
True - for custom networks, we can keep the network name visible.